sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -143,10 +143,10 @@ class SparkSqlAstBuilder extends AstBuilder {
   override def visitExplain(ctx: ExplainContext): LogicalPlan = withOrigin(ctx) {
     val options = ctx.explainOption.asScala
     if (options.exists(_.FORMATTED != null)) {
-      logWarning("EXPLAIN FORMATTED option is ignored.")
+      logWarning("Unsupported operation: EXPLAIN FORMATTED option")
     }
     if (options.exists(_.LOGICAL != null)) {
```

Comment:
I think we don't need to throw an exception here. LOGICAL is more or less the opposite of EXTENDED. So I think we can safely ignore it instead of logging a warning or throwing an exception.